### ISSUE TITLE
mxnet为什么不支持使用conda安装？

### ISSUE THREAD

**Issue by BigBigWolf-AI** [apache/incubator-mxnet#14316]
tensorflow，pytorch都支持conda安装。支持conda安装，对于刚入门的小白来说其实非常好。因为无论是在Windows还是Linux下，都只需要安装显卡驱动就好了。不需要配置cuda和cudnn，因为使用conda install命令安装时，会自动将cuda和cudnn安装好。小白完全不用搞配置路径这些乱七八糟的东西。
最主要是，使用conda创建一个虚拟环境。不同虚拟环境下的conda和cudnn互不干扰，这样可以评估不同cuda版本的影响。而且，使用conda安装，可以将所有的依赖都一并解决，非常好呀。而且还能支持旧系统。更为可怕的是，就算安装CPU版本，也不需要安装MKL库，都是自动下载安装的。强烈建议mxnet增加对conda安装的支持。
可以参考一下conda版的tensorflow的强大之处：
https://www.anaconda.com/tensorflow-in-anaconda/

**Comment by wkcn** [comment]
感谢你的建议: )

pip可以在conda以及非conda环境下使用，pip安装MXNet仅仅需要一行指令。

可以使用conda创建一个虚拟环境，然后使用pip安装MXNet。而且anaconda自身已经带了numpy, opencv这些库了，不需要手动安装额外的库。

MXNet的mkl版本是可以直接使用，不需要安装其他东西的。CUDA版本也是安装好显卡驱动后直接可以使用。

**Comment by piyushghai** [comment]
@mxnet-label-bot Add [Feature Request]

**Comment by BigBigWolf-AI** [comment]
既然都是feature request了，那就再强烈建议mxnet支持一下意法半导体公司的STM32系列单片机，感觉未来在嵌入式前端这一块，会有很大的应用，而且意法半导体也推出了许多相关的产品。支持Keras，tensorflow，cntk都支持，就是不支持mxnet，这怎么行。
参考：
https://www.st.com/content/st_com/en/about/innovation---technology/artificial-intelligence.html

**Comment by wkcn** [comment]
@BigBigWolf-AI 这个是意法半导体公司做的模型格式转换。
现在可以把MXNet的模型以onnx格式导出，再导入支持onnx转换到意法半导体公司的模型格式的框架里，转出最终的模型格式。

**Comment by BigBigWolf-AI** [comment]
好方法，我怎么没有想到呢，66666666666~~~ @wkcn

**Comment by lanking520** [comment]
@BigBigWolf-AI Currently, MXNet has static-linked most of the dependencies including opencv. You should be able to use it with only Python and pip installed. About the edge device support, MXNet support ARM based device and Raspberry Pi.

**Comment by BigBigWolf-AI** [comment]
最新消息是mxnet需要按照cuda。意味着我的系统如果装了cuda 9，那么我就没有办法尝试使用cuda 10版的mxnet。更没办法安装其它支持cuda 10版本的框架。因为系统只装了一个cuda 9. 但是，如果mxnet搞得像pytorch那样使用conda安装时自动安装对应的cuda版本，就可以使用虚拟环境来隔开不同版本的cuda。
话说到这，想问一下，mxnet要使用cuda 10的话，是不是得手动编译安装？
最后：
@lanking520 Thank You！

**Comment by wkcn** [comment]
@BigBigWolf-AI 不需要自己编译。可用pip装mxnet-cu100或mxnet-cu100mkl
